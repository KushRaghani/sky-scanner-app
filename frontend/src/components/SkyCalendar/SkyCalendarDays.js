import React from "react";
import PropTypes from "prop-types";
import currencyUtil from "../../utils/currencyUtil";
import {
  getMonthStart,
  getMonthEnd,
  getWeekStart,
  getWeekEnd,
  isSameMonth,
  formatDate,
  addDays,
  isSameDay
} from "../../utils/dateUtil";

const SkyCalendarDays = ({ currentMonth, selectedDate, data, onDateClick }) => {
  const monthStart = getMonthStart(currentMonth);
  const monthEnd = getMonthEnd(monthStart);
  const startDate = getWeekStart(monthStart);
  const endDate = getWeekEnd(monthEnd);

  const dateFormat = "Day";
  const weeks = [];

  let days = [];
  let day = startDate;
  let formattedDate = "";

  // Keeping a check on the end date for a given selected month
  while (day <= endDate) {
    // Loop over each of the 7 days, starting on a Sunday for a paricular month
    for (let i = 0; i < 7; i++) {
      formattedDate = formatDate(day, dateFormat);
      const cloneDay = day;
      days.push(
        <div
          className={`block cell block-val ${
            !isSameMonth(day, monthStart)
              ? "disabled"
              : isSameDay(day, selectedDate)
              ? "selected"
              : ""
          }`}
          key={day}
          onClick={() => onDateClick(cloneDay)}
        >
          <span className="calendar-date">{formattedDate}</span>
          {data && data[day.toUTCString()] ? (
            <span className="currencyVal">
              {currencyUtil(parseFloat(data[day.toUTCString()]))}
            </span>
          ) : null}
        </div>
      );
      day = addDays(day, 1);
    }
    // Each week is pushed in array of weeks which renders the calendar body
    weeks.push(
      <div className="calendar-row" key={day}>
        {days}
      </div>
    );
    days = [];
  }
  return <div className="main">{weeks}</div>;
};

SkyCalendarDays.propTypes = {
  currentMonth: PropTypes.object.isRequired,
  selectedDate: PropTypes.object.isRequired,
  data: PropTypes.object.isRequired,
  onDateClick: PropTypes.func.isRequired
};

export default SkyCalendarDays;
