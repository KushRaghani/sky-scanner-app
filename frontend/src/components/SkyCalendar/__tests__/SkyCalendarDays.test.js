import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import SkyCalendarDays from "../SkyCalendarDays";

function loadSkyCalendarDays(args) {
  const defaultProps = {
    currentMonth: new Date(),
    selectedDate: new Date(),
    data: {},
    onDateClick: () => {}
  };

  const props = { ...defaultProps, ...args };
  return <SkyCalendarDays {...props} />;
}
describe("SkyCalendarDays", () => {
  // Enyzme tests
  test("renders SkyCalendarDays", () => {
    const wrapper = shallow(loadSkyCalendarDays());
    expect(wrapper.find(".main").length).toBe(1);
    expect(wrapper.find(".calendar-row").length).toBe(5);
    expect(wrapper.find(".calendar-date").length).toBe(35);
  });

  // Snapshot testing
  test("snapshot with current Date and current month", () => {
    const tree = renderer.create(loadSkyCalendarDays());
    expect(tree).toMatchSnapshot();
  });
});
