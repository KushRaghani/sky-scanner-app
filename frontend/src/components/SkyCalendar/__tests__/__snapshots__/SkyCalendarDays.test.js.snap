// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkyCalendarDays snapshot with current Date and current month 1`] = `
<div
  className="main"
>
  <div
    className="calendar-row"
  >
    <div
      className="block cell block-valdisabled"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        28
      </span>
    </div>
    <div
      className="block cell block-valdisabled"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        29
      </span>
    </div>
    <div
      className="block cell block-valdisabled"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        30
      </span>
    </div>
    <div
      className="block cell block-valdisabled"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        31
      </span>
    </div>
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        1
      </span>
    </div>
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        2
      </span>
    </div>
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        3
      </span>
    </div>
  </div>
  <div
    className="calendar-row"
  >
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        4
      </span>
    </div>
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        5
      </span>
    </div>
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        6
      </span>
    </div>
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        7
      </span>
    </div>
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        8
      </span>
    </div>
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        9
      </span>
    </div>
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        10
      </span>
    </div>
  </div>
  <div
    className="calendar-row"
  >
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        11
      </span>
    </div>
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        12
      </span>
    </div>
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        13
      </span>
    </div>
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        14
      </span>
    </div>
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        15
      </span>
    </div>
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        16
      </span>
    </div>
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        17
      </span>
    </div>
  </div>
  <div
    className="calendar-row"
  >
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        18
      </span>
    </div>
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        19
      </span>
    </div>
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        20
      </span>
    </div>
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        21
      </span>
    </div>
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        22
      </span>
    </div>
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        23
      </span>
    </div>
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        24
      </span>
    </div>
  </div>
  <div
    className="calendar-row"
  >
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        25
      </span>
    </div>
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        26
      </span>
    </div>
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        27
      </span>
    </div>
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        28
      </span>
    </div>
    <div
      className="block cell block-valselected"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        29
      </span>
    </div>
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        30
      </span>
    </div>
    <div
      className="block cell block-val"
      onClick={[Function]}
    >
      <span
        className="calendar-date"
      >
        31
      </span>
    </div>
  </div>
</div>
`;
